### PR TITLE
Simplify mapping compilation

### PR DESCRIPTION
**Pull request by chesterbot01**
This PR simplifies the generation of mappings. 
There will be a followup PR to clean up methods that are no longer used.

### CODE REVIEW

File: data/integrations/integrations.yml [hunk 1]
```diff
@@ -2,3 +2,6 @@
 - name: google
   available_versions:
     - google/2021-09-21
+- name: shopify
+  available_versions:
+    - shopify/2022-02
```
Comment (jeanrick) [inline data/integrations/integrations.yml:7]: We should leave this entry for https://github.com/Shopify/product-taxonomy/pull/127 which adds this mapping

File: schema/schema.cue [hunk 2]
```diff
@@ -62,7 +62,7 @@ mappings!: [
 		rules!: [
 			...{
 				input!: {
-					product_category_id: string & =~#category_gid_regex
+					product_category_id: string
```
Comment (jeanrick) [inline schema/schema.cue:65]: This is fine for now until we add a means for the schema to be defined for each input/output type 🤔

File: app/serializers/source_data/product_serializer.rb [hunk 3]
```diff
@@ -18,6 +18,7 @@ def deserialize(payload, product_type)
       if product_type == "ShopifyProduct"
         properties = payload.delete("attributes")
         payload["properties"] = properties
+        payload["product_category_id"] = "gid://shopify/TaxonomyCategory/#{payload["product_category_id"]}"
```
Comment (elsom25) [inline app/serializers/source_data/product_serializer.rb:21]: Any reason this can't come from `Category#gid`?

Comment (jeanrick) [inline app/serializers/source_data/product_serializer.rb:21]: _nit_ Would be nice to have a reusable method for this. Like `Category.gid`

Does this gid work for the old taxonomy @elsom25?

**Review by jeanrick — APPROVED** [review]

File: app/serializers/source_data/product_serializer.rb (hunk 3 above)
Comment (chesterbot01) [reply app/serializers/source_data/product_serializer.rb:21]: Shopify/2022-02's category nodes are not seeded

File: app/services/mapping_builder.rb [hunk 4]
```diff
@@ -2,6 +2,16 @@
 
 class MappingBuilder
   class << self
+    def simple_mapping(mapping_rules:)
+      mapping_rules
+        .map do
+          {
+            input: _1.input.payload.delete_if { |_k, v| v.nil? },
+            output: _1.output.payload.delete_if { |_k, v| v.nil? },
```
Comment (elsom25) [inline app/services/mapping_builder.rb:10]:
```suggestion
            input: _1.input.payload.compact,
            output: _1.output.payload.compact,
```

**Review by elsom25 — APPROVED** [review]
Two small tune-ups but broadly okay for now. Thanks for getting this together quickly

File: app/serializers/source_data/product_serializer.rb (hunk 3 above)
Comment (jeanrick) [reply app/serializers/source_data/product_serializer.rb:21]: Ah, I see there's already a similar comment thread: https://github.com/Shopify/product-taxonomy/pull/134/files#r1563212184

File: data/integrations/integrations.yml (hunk 1 above)
Comment (chesterbot01) [reply data/integrations/integrations.yml]: removed.

**Pull request merged**